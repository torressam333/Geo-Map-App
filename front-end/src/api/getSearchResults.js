import axios from 'axios';

const getSearchResults = async (
  searchQuery,
  coords = '0,0',
  long = '0',
  lat = '0'
) => {
  try {
    if (searchQuery !== '') {
      // Create params object
      const params = new URLSearchParams({
        fuzzyMatch: true,
        language: 'en',
        limit: 10,
        proximity: coords ? `${long},${lat}` : '0,0',
      });

      const response = await axios.get(
        `http://localhost:3000/api/search/${searchQuery}?${params}`
      );

      console.log(response.data.features);

      return response?.data?.features;
    }
  } catch (err) {
    throw new Error(err.message);
  }
};

export default getSearchResults;
