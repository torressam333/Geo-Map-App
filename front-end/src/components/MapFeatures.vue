<template>
  <section
    class="w-full md:w-auto absolute md:top-[40px] md:left-[60px] z-[2] flex gap-4 px-6 py-8 md:px-0 md:py-0 bg-transparent"
  >
    <div
      class="px-4 bg-white flex items-center shadow-md rounded-md min-h-[50px]"
    >
      <i class="fas fa-location-arrow text-slate-700 text-[1.5rem]"></i>
    </div>
  </section>
</template>

<script>
export default {
  name: 'MapFeatures',
};
</script>
