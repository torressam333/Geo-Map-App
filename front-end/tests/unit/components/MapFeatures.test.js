import { shallowMount } from '@vue/test-utils';
import MapFeatures from '@/components/MapFeatures';
import axios from 'axios';
import getSearchResults from '@/api/getSearchResults';

// Let's mock this bad boy
jest.mock('axios');

describe('MapFeatures', () => {
  let wrapper;

  beforeEach(() => {
    // Make it anew for each test
    axios.mockClear();

    jest.useFakeTimers('legacy');

    // Fake api data
    axios.get.mockResolvedValue({
      searchQuery: 'Starbucks',
      params: {
        fuzzyMatch: true,
        language: 'en',
        limit: 10,
      },
    });

    wrapper = shallowMount(MapFeatures, {
      propsData: {
        coords: {},
        fetchCoords: true,
      },
    });
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('emits a method to toggle location services', async () => {
    wrapper.vm.$emit('getGeoLocation');

    // Wait until $emits have been handled
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().getGeoLocation).toBeTruthy();
  });

  it('renders the modal markup correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should clear timeout after fetching data', async () => {
    const setTimeout = jest.fn();

    setTimeout(1500);

    // Mock api call after timeout expires
    await getSearchResults();

    // Assertions
    expect(setTimeout).toHaveBeenCalled();
    expect(axios.get).toHaveBeenCalled();
  });
});
