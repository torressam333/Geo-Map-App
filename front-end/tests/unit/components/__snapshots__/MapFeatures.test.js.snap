// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapFeatures renders the modal markup correctly 1`] = `
<section
  class="w-full md:w-auto absolute md:top-[40px] md:left-[60px] z-[2] flex gap-2 px-6 py-8 md:px-0 md:py-0 bg-transparent"
>
  <!-- Location Searching -->
  <div
    class="relative flex-1 md:min-w-[350px]"
  >
    <!-- search input -->
    <input
      aria-label="Search for location"
      class="pl-8 pr-4 py-3 text-sm focus:outline-none shadow-md rounded-md w-full"
      placeholder="Search for location"
      type="text"
    />
    <!-- Search icon -->
    <div
      class="absolute top-0 left-[10px] h-full flex items-center"
    >
      <i
        class="fas fa-search"
      />
    </div>
    <search-results-stub />
  </div>
  <!-- Geolocation -->
  <div
    class="px-4 bg-white flex items-center shadow-md rounded-md min-h-[50px] bg-slate-600"
  >
    <i
      class="fas fa-location-arrow text-[1.5rem] cursor-pointer text-white animate-pulse"
    />
  </div>
</section>
`;
